src/victory-util/data.js

```diff
@@ -63,40 +83,83 @@ export default {
    * @returns {Array} the formatted data
    */
   formatData(dataset, props, stringMap) {
-    if (!Array.isArray(dataset)) {
+    const isArrayOrIterable = Array.isArray(dataset) || Immutable.isIterable(dataset);
+    if (!isArrayOrIterable) {
       return [];
     }
+
     stringMap = stringMap || {
       x: this.createStringMap(props, "x"),
       y: this.createStringMap(props, "y")
     };
+
     const accessor = {
       x: Helpers.createAccessor(props.x !== undefined ? props.x : "x"),
       y: Helpers.createAccessor(props.y !== undefined ? props.y : "y"),
       y0: Helpers.createAccessor(props.y0 !== undefined ? props.y0 : "y0")
     };
-    const data = dataset.map((datum, index) => {
-      const evaluatedX = datum._x !== undefined ? datum._x : accessor.x(datum);
-      const evaluatedY = datum._y !== undefined ? datum._y : accessor.y(datum);
-      const y0 = datum._y0 !== undefined ? datum._y0 : accessor.y0(datum);
+
+    const data = dataset.reduce((dataArr, datum, index) => { // eslint-disable-line complexity
+      const trimmedDatum = Array.isArray(datum) || Immutable.isList(datum)
+        ? datum
+        : this.trimDatum(datum, props);
```

Comment:
another potential strategy would be to do a "shallow" `toJS` conversion (pull out all the top level properties into an object but don't convert those objects using `toJS`). this way, any props passed in will remain available.

Reply:
I think this might be a safer solution.

Reply:
Alright, I'll try it out with that solution.